packages/synthetics-sdk-broken-links/src/navigation_func.ts

```diff
@@ -77,6 +80,54 @@ export async function retrieveLinksFromPage(
   );
 }
 
+/**
+ * Concurrently checks a list of links on multiple Puppeteer pages, records the results,
+ * and throws a detailed error if necessary.
+ *
+ * @param browser - The Puppeteer browser instance used to create pages.
+ * @param links - An array of links to check.
+ * @param options - global options object with all broken link checker options.
+ * @returns A Promise that resolves with an array of successfully checked link
+ *          results
+ * @throws {Error} If an error occurs while checking the links, it throws an
+ *          error with an appropriate message, including which link errored.
+ */
+export async function checkLinks(
+  browser: Browser,
+  links: LinkIntermediate[],
+  options: BrokenLinksResultV1_BrokenLinkCheckerOptions
+): Promise<BrokenLinksResultV1_SyntheticLinkResult[]> {
+  const followed_links: BrokenLinksResultV1_SyntheticLinkResult[] = [];
+  const limit = pLimit(options.link_limit! / 10);
+
+  // Initialize a pool of pages
+  const pagePool: Page[] = [];
+
+  await Promise.all(
```

Comment:
I think that we talked about this in another thread - im still confused as to what happens if any of these links end up failing? Doesnt this exit mid-check without going through all the links?

I'm thinking through scenarios where for some reason, 99% of links work except 1 which crashes puppeteer for some reason. Do we really want to abandon ship and not return that information? LMK if im missing something

Reply:
Yes and no...

- The call to page.goto(), which is the actual navigation is done in a function called `fetchLink()`, this has its own try-catch which handles if an error is thrown.

- checkLink() has no other asynchronous call, the only "risky" call is to page.goto(). `checkLink()` should never throw an unexpected error and this catch block should never be executed, this is just incase. Also I don't think  there is a  way to handle this error other than throwing or ignoring and logging (which is not severe enough for the scenario in my opinion).